plasmapy/formulary/shocks.py

```diff
@@ -0,0 +1,116 @@
+""" This module gathers functions relating to shocks and the properties thereof.
+"""
+__all__ = ["entropy_across_shock_polytropic"]
+
+
+import warnings
+from astropy import units as u
+from plasmapy.utils.exceptions import PhysicsWarning
+from plasmapy.utils.decorators import (
+    angular_freq_to_hz,
+    check_relativistic,
+    validate_quantities,
+)
+from astropy.constants import c, a0, k_B
+from numpy import pi, exp, sqrt, log, nan, isclose
+
+
+@validate_quantities(
+    rho_1={"can_be_negative": False, "pass_equivalent_units": True},
+    rho_2={"can_be_negative": False, "pass_equivalent_units": True},
+    p_1={"can_be_negative": True, "pass_equivalent_units": True},
+    p_2={"can_be_negative": True, "pass_equivalent_units": True},
+    c_v={"can_be_negative": False},
+)
+def entropy_across_shock_polytropic(
```

Comment:
This is a hard function to name!  While current name is okay, I still feel it's a little vague.  How about something like `entropy_jump_polytropic`, `entropy_jump_polytropic_shock`, `entropy_jump_polytropic_gas_shock`, etc.  I moved to the "jump" nomenclature, because in the shock world the jump conditions refers to the relations between the pre- and post-shock.  I think I'm leaning towards `entropy_jump_polytropic_gas_shock`, but there's probably still room for improvement.